### ISSUE TITLE
Output directory must already exist

### ISSUE THREAD

**Issue by skhilton** [jbloomlab/pdb_prot_align#2]
If the output directory does not already exist, you get an error like 
```
Running `pdb_prot_align` 0.1.0

Parsing PDB input_files/4O5N_trimer.pdb chains A C E
Parsed sequence of 317 residues, ranging from 9 to 325 in PDB numbering.

Read 3 sequences from input_files/HA.fasta
Reference protein is of length 564 and has the following header:
cds:BAA14332 A/duck/Czechoslovakia/1956 1956// HA

Using `mafft` to align sequences to output_files/HA_unstripped.fasta
Traceback (most recent call last):
  File "/Users/skhilton/miniconda3/bin/pdb_prot_align", line 10, in <module>
    sys.exit(_run_main())
  File "/Users/skhilton/miniconda3/lib/python3.7/site-packages/pdb_prot_align/main.py", line 34, in _run_main
    run(**kwargs)
  File "/Users/skhilton/miniconda3/lib/python3.7/site-packages/pdb_prot_align/main.py", line 193, in run
    Bio.AlignIO.write(aln, alignment_unstripped, 'fasta')
  File "/Users/skhilton/miniconda3/lib/python3.7/site-packages/Bio/AlignIO/__init__.py", line 220, in write
    with as_handle(handle, "w") as fp:
  File "/Users/skhilton/miniconda3/lib/python3.7/contextlib.py", line 112, in __enter__
    return next(self.gen)
  File "/Users/skhilton/miniconda3/lib/python3.7/site-packages/Bio/File.py", line 120, in as_handle
    with open(handleish, mode, **kwargs) as fp:
FileNotFoundError: [Errno 2] No such file or directory: 'output_files/HA_unstripped.fasta'
```

This error goes away if I `mkdir output_files` before I run the `pdb_prot_align` command. 

Let me know if you would like some small example files.

**Comment by jbloom** [comment]
It's debatable whether this is a bug or not, as many programs require output directories to already be produced before they run. But I certainly agree that if we require that, we should give a more informative error message.

So I would suggest modifying the relevant code (which is in `main.run`) to do the following:

1. Get the output directory(ies) name (if present) by `os.path.dirname(alignment)` and `os.path.dirname(csv)`.

2. Then **either** create the directories with `os.makedirs(<dirname>, exist_ok=True)` which can be safely run even if directory exists **or** raise an informative error saying directory needs to be created first. Which of these we do depends on whether the "correct" behavior is creating directory or raising error. Maybe the former is better?

@skhilton: Do you want to do this and submit pull request?

**Comment by jbloom** [comment]
Fixed in pull request #4
